src/main/java/org/apache/commons/compress/archivers/dump/DumpArchiveEntry.java

```diff
@@ -418,12 +418,8 @@ public boolean equals(final Object o) {
         }
 
         // summary is always null right now, but this may change some day
-        if ((summary == null && rhs.summary != null) // NOSONAR
-            || (summary != null && !summary.equals(rhs.summary))) { // NOSONAR
-            return false;
-        }
-
-        return true;
+        return (summary != null || rhs.summary == null) // NOSONAR
+                && (summary == null || summary.equals(rhs.summary));
     }
```

Comment:
I find this version a lot harder to read than the original expression. I'm totally fine with getting rid of the `if`, but please keep the original logic.

Reply:
I've double checked and reverted to the original logic.